pkg/kubelet/apis/config/validation/validation.go

```diff
@@ -114,6 +114,16 @@ func ValidateKubeletConfiguration(kc *kubeletconfig.KubeletConfiguration) error
 	if kc.TopologyManagerPolicy != kubeletconfig.NoneTopologyManagerPolicy && !localFeatureGate.Enabled(features.TopologyManager) {
 		allErrors = append(allErrors, fmt.Errorf("invalid configuration: TopologyManager %v requires feature gate TopologyManager", kc.TopologyManagerPolicy))
 	}
+	for k, v := range kc.KubeReserved {
```

Comment:
You need to check for other parse errors in the quantity.  Look at https://github.com/kubernetes/kubernetes/blob/master/cmd/kubelet/app/server.go#L1177-L1183

Reply:
Thanks for the feedback. I have added a new commit.

Reply:
Note that https://github.com/kubernetes/kubernetes/blob/master/cmd/kubelet/app/server.go#L1173 only checks certain resources (and disallows everything else).

It would be worth considering whether the current `parseResourceList` should be used in both places, so we don't duplicate code of this kind.

Reply:
It appears I would have to modify the `server.go` and `server_test.go` to export `ParseResourceList` so that I would be able to access it. Would that be okay?

Reply:
Hmm, curious to hear from @RobertKrawitz but I don't like the idea of `pkg/kubelet` importing from `cmd/kubelet`. 

One option would be to move the code that is currently in `cmd/kubelet` into `pkg/kubelet`, and then both can import it. However, I'm not sure if other dependencies will come into play and make that more difficult.

Regardless, I think using `ParseQuantity` is a positive change - thanks :)